adds changelog for vite plugin (#21490)
cloudflare/cloudflare-docs: src/content/changelog/workers/2025-04-08-vite-plugin.mdx

---

> title: The Cloudflare Vite plugin is now generally available

Reviewer: ```suggestion
title: The Cloudflare Vite plugin is now Generally Available
```

description: Cloudflare Vite Plugin 1.0 is out now — with official support for Vite 6, React Router v7, and the full Developer Platform. Now you can run Vite’s dev server locally, while executing your code directly in the Workers runtime!
products:
  - workers
date: 2025-04-08T18:00:00Z
---

> We're excited to announce that the official [Cloudflare Vite plugin](/workers/vite-plugin/) has [reached v1.0](https://blog.cloudflare.com/introducing-the-cloudflare-vite-plugin) and is now **generally available**. The Vite plugin allows you to run [Vite's](https://vite.dev/) development server in the Workers runtime (`workerd`), meaning you get all the benefits of Vite (including [Hot Module Replacement](https://vite.dev/guide/features.html#hot-module-replacement)) while still being able to use features that are exclusive to Workers ([like Durable Objects](/durable-objects/)).

Reviewer: ```suggestion
The [Cloudflare Vite plugin](/workers/vite-plugin/) has [reached v1.0](https://blog.cloudflare.com/introducing-the-cloudflare-vite-plugin) and is now Generally Available ("GA").

When you use `@cloudflare/vite-plugin`, you can use Vite's local development server and build tooling, while ensuring that while developing, your code runs in [`workerd`](https://github.com/cloudflare/workerd), the open-source Workers runtime.

This lets you get the best of both worlds for a full-stack app — you can use [Hot Module Replacement](https://vite.dev/guide/features.html#hot-module-replacement) from Vite right alongside [Durable Objects](/durable-objects/) and other runtime APIs and bindings that are unique to Cloudflare Workers.
```

> This effort was done [🧡 in partnership with the Vite team 🧡](https://blog.cloudflare.com/introducing-the-cloudflare-vite-plugin) and made by possible with their new [environment API](https://vite.dev/guide/api-environment).

Reviewer: ```suggestion
`@cloudflare/vite-plugin` is made possible by the new [environment API](https://vite.dev/guide/api-environment) in Vite, and was built [in partnership with the Vite team](https://blog.cloudflare.com/introducing-the-cloudflare-vite-plugin).
```

#### Framework support

> The Vite plugin is supported in front-end [frameworks](/workers/frameworks/) (for building single page applications and static sites) as well as [React Router v7](/workers/frameworks/framework-guides/react-router/) (for building server-side rendered applications). For an example, check out [this full-stack application](https://github.com/cloudflare/templates/tree/main/react-router-postgres-ssr-template) built using React Router v7 and the Vite plugin.

Reviewer: ```suggestion
You can build any type of application with `@cloudflare/vite-plugin`, using any rendering mode, from single page applications (SPA) and static sites to server-side rendered (SSR) pages and API routes.

[React Router v7 (Remix)](/workers/frameworks/framework-guides/react-router/) is the first full-stack framework to provide full support for Cloudflare Vite plugin, allowing you to use all parts of Cloudflare's developer platform, without additional build steps.
```

> You can also build complete full-stack apps on Workers **without a framework**: ["just use Vite"](https://blog.cloudflare.com/introducing-the-cloudflare-vite-plugin) and React together, and build a back-end API in the same Worker. Follow our [React SPA with an API tutorial](/workers/vite-plugin/tutorial/) to learn how.

Reviewer: ```suggestion
You can also build complete full-stack apps on Workers **without a framework** — ["just use Vite"](https://blog.cloudflare.com/introducing-the-cloudflare-vite-plugin) and React together, and build a back-end API in the same Worker. Follow our [React SPA with an API tutorial](/workers/vite-plugin/tutorial/) to learn how.
```

---
title: The Cloudflare Vite plugin is now generally available
description: Cloudflare Vite Plugin 1.0 is out now — with official support for Vite 6, React Router v7, and the full Developer Platform. Now you can run Vite’s dev server locally, while executing your code directly in the Workers runtime!
products:
  - workers
date: 2025-04-08T18:00:00Z

> hidden: true

Reviewer 2: ```suggestion
```

---

We're excited to announce that the official [Cloudflare Vite plugin](/workers/vite-plugin/) has [reached v1.0](https://blog.cloudflare.com/introducing-the-cloudflare-vite-plugin) and is now **generally available**. The Vite plugin allows you to run [Vite's](https://vite.dev/) development server in the Workers runtime (`workerd`), meaning you get all the benefits of Vite (including [Hot Module Replacement](https://vite.dev/guide/features.html#hot-module-replacement)) while still being able to use features that are exclusive to Workers ([like Durable Objects](/durable-objects/)).

This effort was done [🧡 in partnership with the Vite team 🧡](https://blog.cloudflare.com/introducing-the-cloudflare-vite-plugin) and made by possible with their new [environment API](https://vite.dev/guide/api-environment).

#### Framework support

> The Vite plugin is supported in frontend [frameworks](/workers/frameworks/) (for building single-page applications and static sites) as well as [React Router v7](/workers/frameworks/framework-guides/react-router/) (for building server-side rendered applications). For an example, check out [this full-stack application](https://github.com/cloudflare/templates/tree/main/react-router-postgres-ssr-template) built using React Router v7 and the Vite plugin.

Reviewer 3: ```suggestion
The Vite plugin is supported in front-end [frameworks](/workers/frameworks/) (for building single page applications and static sites) as well as [React Router v7](/workers/frameworks/framework-guides/react-router/) (for building server-side rendered applications). For an example, check out [this full-stack application](https://github.com/cloudflare/templates/tree/main/react-router-postgres-ssr-template) built using React Router v7 and the Vite plugin.
```

> You can also build complete full-stack apps on Workers **without a framework**:[“just use Vite"](https://blog.cloudflare.com/introducing-the-cloudflare-vite-plugin) and React together, and build a backend API in the same Worker. Follow our [React SPA with an API tutorial](/workers/vite-plugin/tutorial/) to learn how.

Reviewer 3: ```suggestion
You can also build complete full-stack apps on Workers **without a framework**: ["just use Vite"](https://blog.cloudflare.com/introducing-the-cloudflare-vite-plugin) and React together, and build a back-end API in the same Worker. Follow our [React SPA with an API tutorial](/workers/vite-plugin/tutorial/) to learn how.
```

#### Configuration

> If you're already using [Vite](https://vite.dev/) as your preferred build and development tooling, you can keep using Vite and build applications on Workers — with minimal configuration in your `vite.config.ts`:

Reviewer 3: ```suggestion
If you're already using [Vite](https://vite.dev/) in your build and development toolchain, you can start using our plugin with minimal changes to your `vite.config.ts`:
```

```ts title="vite.config.ts"
import { defineConfig } from "vite";
import { cloudflare } from "@cloudflare/vite-plugin";

export default defineConfig({
	plugins: [cloudflare()],
});
```

> Get started with the Cloudflare Vite plugin by reading our [documentation](/workers/vite-plugin/)!

Reviewer 3: ```suggestion
Take a look at the [documentation for our Cloudflare Vite plugin](/workers/vite-plugin/) for more information!
```
